### DISCUSSION TITLE
How to project MultiBlock onto a plane

### DISCUSSION THREAD

**Discussion by nimafo** [pyvista/pyvista#3876] [Ideas]
Hi,

I need to project a multiblock onto a plane, just like 'project_points_to_plane' function.
How can this be done? I tried 'prject_to_plane' but it only works on meshes? (error: 'MultiBlock' object has no attribute 'project_points_to_plane')

Any idea?

Best,
N.

**Comment by MatthewFlamm** [comment]
As you said, it only works on meshes.  MultiBlock is a collection of meshes.  You can iterate over the internal meshes and form a new MultiBlock, or use `inplace=True` to modify the existing meshes.

```py
import pyvista as pv
mesh = pv.MultiBlock([pv.Sphere(), pv.Cone(center=(0., 0., 5.))])
proj_mesh = pv.MultiBlock([m.project_points_to_plane(origin= (0., 0., 0.)) for m in mesh])
print(proj_mesh)
```
```
MultiBlock (0x7f3698c21820)
  N Blocks:     2
  X Bounds:     -0.500, 0.500
  Y Bounds:     -0.500, 0.500
  Z Bounds:     0.000, 0.000
```

**Reply by MatthewFlamm** [reply]
Note that the default for this method is to use the origin of the mesh, which will give you different origins for each sub-mesh.  We could probably extend this method to MultiBlock for easier use.
